server/lib/parties/party-service.test.ts

```diff
@@ -0,0 +1,264 @@
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client8 = connector.connectClient(user8, USER8_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client9 = connector.connectClient(user9, USER9_CLIENT_ID)
+
+    clearTestLogs(nydus)
+  })
+
+  describe('invite', () => {
+    let leader: PartyUser
+    let party: PartyRecord
+
+    describe('when party exists', () => {
+      beforeEach(() => {
+        leader = user1
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user2])
+        partyService.acceptInvite(party.id, user2, USER2_CLIENT_ID)
+      })
+
+      test('should throw if invited by non-leader', () => {
+        const result = () => partyService.invite(user2, USER2_CLIENT_ID, [user3])
+
+        expect(result).toThrow()
+      })
+
+      test('should update the party record', () => {
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user3])
+
+        expect(party.invites).toMatchObject(new Map([[user3.id, user3]]))
+      })
+
+      test('should publish "invite" message to the party path', () => {
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user3])
+
+        expect(nydus.publish).toHaveBeenCalledWith(getPartyPath(party.id), {
+          type: 'invite',
+          invites: [user3],
+        })
+      })
+    })
+
+    describe("when party doesn't exist", () => {
+      beforeEach(() => {
+        leader = user1
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user2, user3])
+      })
+
+      test('should create a party record', () => {
+        expect(party).toMatchObject({
+          id: party.id,
+          invites: new Map([
+            [user2.id, user2],
+            [user3.id, user3],
+          ]),
+          members: new Map([[leader.id, leader]]),
+          leaderId: leader.id,
+        })
+      })
+
+      test('should subscribe leader to the party path', () => {
+        expect(client1.publish).toHaveBeenCalledWith(getPartyPath(party.id), {
+          type: 'init',
+          party,
+        })
+      })
+    })
+
+    test('should subscribe invited users to the invites path', () => {
+      leader = user1
+      party = partyService.invite(leader, USER1_CLIENT_ID, [user2, user3])
+
+      expect(client2.publish).toHaveBeenCalledWith(getInvitesPath(party.id), {
+        type: 'invite',
+        from: party.leaderId,
+      })
+      expect(client3.publish).toHaveBeenCalledWith(getInvitesPath(party.id), {
+        type: 'invite',
+        from: party.leaderId,
+      })
+    })
+  })
+
+  describe('removeInvite', () => {
+    let leader: PartyUser
+    let party: PartyRecord
+
+    beforeEach(() => {
+      leader = user1
+      party = partyService.invite(leader, USER1_CLIENT_ID, [user2, user3])
+    })
+
+    test('should throw if the party is not found', () => {
+      const result = () => partyService.removeInvite('INVALID_PARTY_ID', user2)
+
+      expect(result).toThrow()
```

Comment:
Also same as above, if we have a particular type of error we expect based on the API, try to match on that when using this